models/contrib/nish/abracadabra.py

```diff
@@ -0,0 +1,423 @@
+    borrow_fee : float
+    interest : float
+
+
+class AbracadabraPortfolio(IterableListGenericDTO[AbracadabraVaultPortfolio]):
+    abracadabra_portfolio: List[AbracadabraVaultPortfolio]
+    _iterator: str = 'abracadabra_portfolio'
+
+
+
+# Fetching Collateral of each market of abracadabra on ethereum chain
+@Model.describe(slug="contrib.abracadabra-vault-portolio",
+                version="1.0",
+                display_name="Get TVL for abracadabra",
+                description="Get TVL for abracadabra",
+                input=Contract,
+                output=AbracadabraVaultPortfolio)
+class AbracadabraGetVaultPortfolio(Model):
+    def run(self, input) -> AbracadabraVaultPortfolio:
+        # Keys of ethereum_active_markets
+        ethereum_active_markets_keys = list(ethereum_active_markets.keys())
+        # Values of ethereum_active_markets
+        ethereum_active_markets_vaules = list(ethereum_active_markets.values())
+        ethereum_active_markets_vaules = [value.lower() for value in ethereum_active_markets_vaules]
+        # Name of vault
+        position = ethereum_active_markets_vaules.index(str(input.address).lower())
+        vault_name = ethereum_active_markets_keys[position]
+
+        # Contract address of market
+        market_address = Address(input.address).checksum
+
+        # Contract Instance
+        market_contract = Contract(address=market_address, abi=ABRACADABRA_CAULDRON_ABI)
+
+        # Borrow Fee
+        borrow_fee = try_or(
+            lambda: float(market_contract.functions.BORROW_OPENING_FEE().call()) / pow(10,3)
+        )
+        # Maximum Colletaral Ratio
+        maximum_collateral_ratio = try_or(
+            lambda: float(market_contract.functions.COLLATERIZATION_RATE().call()) / pow(10,3)
+        )
+        # Liquidation Fee
+        liquidation_fee = try_or(
+            lambda: float(
+                        str(market_contract.functions.LIQUIDATION_MULTIPLIER().call())[1:]
+                    ) / pow(10,3)
+        )
+        # Interest
+        interest_ps = try_or(
+            lambda: float(market_contract.functions.accrueInfo().call()[2]) / pow(10,3)
+        )
+        interest = try_or(lambda: interest_ps / (60*60*24*365))
+
+        # Contract address of collateral
+        collateral = Token(
+            address = Address(market_contract.functions.collateral().call()).checksum)
+        # Decimals in collateral
+        decimals = collateral.decimals
+        # Token Price
+        _exchange_rate = float(market_contract.functions.exchangeRate().call())
+        exchange_rate = 1 / _exchange_rate * pow(10,decimals)
+        collateral_deposited = float(
+            market_contract.functions.totalCollateralShare().call()) / pow(10,decimals)
+        collateral_value = collateral_deposited * exchange_rate
+        mim_borrowed = float(
+            market_contract.functions.totalBorrow().call()[1]) / pow(10,18)
+
+
+        return AbracadabraVaultPortfolio(
```

Comment:
How do we make this the same structure as aave adn compound? (question for later. ask @leafyoung about this)

Reply:
@nishgaba-ai  let me know when this is fixed